specifications/secure-launch-specification.rst

```diff
@@ -0,0 +1,719 @@
+:dl_handler: The address to the entry point for the DLE Handler.
+:dce_base: The base address where the DCE is located.
+:dce_size: The size of the DCE.
+:dlme_entry: The address for the entry point of the DLME.
+
+.. code-block:: c
+    :linenos: 1
+
+    struct slr_entry_dl_info {
+        struct slr_entry_hdr hdr;
+        struct slr_bl_context bl_context;
+        u64 dl_handler;
+        u64 dce_base;
+        u32 dce_size;
+        u64 dlme_entry;
+    };
+
+Boot Loader Context
+"""""""""""""""""""
+
+There may be a situation where the bootloader will need to leave a context
+object containing platform specific information or helper callbacks for the DLE
+Handler to use.
+
+.. note::
+   It is out-of-scope for this specification, but it is advised that if a
+   platform or a bootloader requires the use of a context object, they should
+   standardize their context object to enable independent DLE Handlers
+
+:bootloader: An identifier the bootloader should use for ident and version.
+:context: Address of the context object.
+
+.. code-block:: c
+    :linenos: 1
+
+    struct slr_bl_context {
+        u16 bootloader;
+        u16 reserved;
+        u64 context;
+    };
+
+DRTM TPM Event Log
+^^^^^^^^^^^^^^^^^^
+
+:REQUIRED: This entry **MUST** be present.
+
+This entry describes where and what type of TPM event log should be used.
+
+.. note::
+   On most platforms, the TCG UEFI TPM event log format should be used. The
+   ability to specify alternatives is to support older platforms that are not
+   aware of the modern event log format or can support multiple formats.
+
+:tag: SLR_ENTRY_LOG_INFO
+:format: The type of TPM event log format to use.
+:addr: The base address where the log should reside.
+:size: The size allocated for the log.
+
+.. code-block:: c
+    :linenos: 1
+
+    struct slr_entry_log_info {
+        struct slr_entry_hdr hdr;
+        u16 format;
+        u16 reserved;
+        u64 addr;
+        u32 size;
+    };
+
+DRTM Measurement Policy
+^^^^^^^^^^^^^^^^^^^^^^^
+
+:REQUIRED: This entry **MUST** be present and have the required entries.
+
+The measurement policy is for conveying to the SL Entry on what it should
+measure, where that entity is located, which PCR the measurement should be
+stored, and how the event should be identified in the TPM event log. 
+
+.. warning::
+   The SL Entry **SHALL** fail if it determines an invalid policy is present.
+
+:tag: SLR_ENTRY_ENTRY_POLICY
+:revision: A revision field to identify the version of policy being used.
+:nr_entries: The total number of policy entries available.
+
+.. code-block:: c
+    :linenos: 1
+
+    struct slr_entry_policy {
+        struct slr_entry_hdr hdr;
+        u16 revision;
+        u16 nr_entries;
+        /* policy_entries[] */
+    };
+  
+
+DRTM Policy Entry
+"""""""""""""""""
+
+A policy entry represents an entity that the SL Entry is being requested to
+measure. As an SL Entry is able to measure an attribute of the launch
```

Comment:
What is an attribute of the launch environment?

Reply:
Any characteristic, property, or component of the launch environment.

Reply:
I had to think about this, and upon looking at the definition, this is the specific definition I was intending with this use,
_an object closely associated with or belonging to a specific person, thing, or office_

As for an example, it could be possible to create a policy entry requesting a hash of all PCI device IDs of devices attached at boot.

Reply:
> As for an example, it could be possible to create a policy entry requesting a hash of all PCI device IDs of devices attached at boot.

And this translates to something meaningful, but when I read attributes as "Any characteristic, property, or component of the launch environment." it is an abstract concept. I have no idea how to attach that to reality. If we consider the reader aware of how those abstracts materialize, that's ok, but you can see that I should be somehow familiar with what was written, and despite that, it wasn't evident to me. So using an example, IMO, makes a lot of sense for this spec.

Finally, it came to me while reviewing Fobnail documentation we should always give the shortest possible path so the developer/user/reader can see how to use a given tool. In this case, we define some structures, so let's ensure those who use them will understand how they should be used. Even RFC has sections Examples to show how concept should be used in practice, e.g., https://www.rfc-editor.org/rfc/rfc3629#page-8

Reply:
The spec defining this needs to remain generic so as not inhibit implementers from using it in a manner that meets their needs. I have no issue with an appendix being added later that contains policy examples.

Reply:
RFCs have different approaches since there are dedicated chapters, for example, the use of specification, but if there is consensus (I see @krystian-hebel :+1: ) for the appendix, then I'm ok with resolving this discussion.